### ISSUE TITLE
Sobel filter should be 2d for images

### ISSUE THREAD

**Issue by Brow71189** [nion-software/niondata#12]

**Comment by cmeyer** [comment]
Sobel currently uses `numpy.ndimage.sobel`. This looks like it applies a 2D filter. What am I missing?

https://docs.scipy.org/doc/scipy/reference/generated/scipy.ndimage.sobel.html

**Comment by Brow71189** [comment]
the default for `axis` is `-1` according to the documentation. To me this looks like it is only applying the filter to the last axis of the image.
Tom (who reported this) also said that you only get the filter in one direction. I'm not sure how exactly he would expect the output to look like, but I'll ask him.
Wikipedia also shows and example for how to calculate sobel of 2D images, so maybe we can use this: [Sobel operator](https://en.wikipedia.org/wiki/Sobel_operator)
I think overall this would be another application for more user configurable options for our built-in processing tools.

**Comment by cmeyer** [comment]
Here's another useful link:

[Stack OverFlow: Applying the Sobel filter using scipy](https://stackoverflow.com/questions/7185655/applying-the-sobel-filter-using-scipy)

**Comment by cmeyer** [comment]
I did some quick experiments with the following code:

```
dx = scipy.ndimage.sobel(r03.data, 0)
dy = scipy.ndimage.sobel(r03.data, 1)
m = numpy.hypot(dx, dy)
m = m / numpy.max(m)
show(m)
```

It seems that the 2D Sobel is useful on some data and not other data. I'm thinking that we should give the user a choice as to what dimension to use (x, y, or radially). We should probably also give them a choice of kernel (type, size).

See https://github.com/nion-software/nionswift/issues/322 too.

**Comment by tcpekin** [comment]
Agree, I think 2D sobel is the most naively useful. Any time you have curved edges I think it is better than the single axis equivalent. Especially since the use case I have seen the most in microscopy is finding edges of circular diffraction disks/spots, the 2D version is definitely better than the rest. Of course the flexibility of being able to choose is the best possibility. 

Another option would be to be able to define any kernel that would then be convolved with the image, with some defaults available, but if you do that, you might be better off just scripting the same thing.
